feat: allow download charts as image,pdf

joaoviana opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

Closes: [#11529](https://github.com/lightdash/lightdash/issues/11529)

### Description:

Allows downloading charts as image,pdf,json.etc - uses the same component as the Explorer

Demo: 

https://github.com/user-attachments/assets/53d131c8-f5e2-4a4a-aab1-771f60dcd61b



### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

joaoviana commented on packages/frontend/src/components/ChartDownload.tsx:
```diff
@@ -215,18 +209,14 @@ const ChartDownloadOptions: React.FC<DownloadOptions> = ({
-                    Download
-                </Button>
-            )}
+            <Button
```
we would never reach this statement. It's moved down below before rendering this component.

joaoviana commented on packages/frontend/src/features/sqlRunner/components/Download/ChartDownload.tsx:
```diff
@@ -0,0 +1,83 @@
+                            />
+                        </Radio.Group>
+                        {downloadFormat === DownloadFileType.IMAGE && (
+                            <ChartDownloadOptions
```
use the same chart download options when choosing `image` download format.

rephus reviewed (approved):
it works great. 

Can we remo the JSON option though ? 

it looks like it is the results in JSON format, but it looks more like the echart configa, and also, it is not an "image" format...
Instead, we could add a proper JSON format , next to CVS and image, to download the results

![image](https://github.com/user-attachments/assets/8bc96875-ac43-4c3b-a79b-949c3d11c252)

joaoviana commented:
> it works great.
> 
> Can we remo the JSON option though ?
> 
> it looks like it is the results in JSON format, but it looks more like the echart configa, and also, it is not an "image" format... Instead, we could add a proper JSON format , next to CVS and image, to download the results (on a separate ticket perhaps)
> 
> ![image](https://private-user-images.githubusercontent.com/1983672/369904859-8bc96875-ac43-4c3b-a79b-949c3d11c252.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxMDA0NjgsIm5iZiI6MTcyNzEwMDE2OCwicGF0aCI6Ii8xOTgzNjcyLzM2OTkwNDg1OS04YmM5Njg3NS1hYzQzLTRjM2ItYTc5Yi05NDljM2QxMWMyNTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjNUMTQwMjQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YmUzYjY1M2FiZWFmNDIxMmZiYjBmMjg4ZDM5NzVkYWQ1MGZiZmU3OTZkZTA4MjY4ZjIyYzJmNGI5Y2Y0YzU4MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ltk7iIu4vOx7QJMmtD096CK6jsA1FKdv3QLe5wkYOpk)

Great point @rephus , removed that option on sql runner charts 👍 

We can add the option later in a separate label when we feel the need from user feedback 🙌 


<img width="265" alt="Screenshot 2024-09-23 at 15 01 16" src="https://github.com/user-attachments/assets/83774e12-c4d8-469c-ab80-75a10ab2f2d0">
Removed in sql runner 🆙
